### PR TITLE
Typescript Migration PR

### PR DESCRIPTION
**Pull request by seanchen0818**
Migrated all working js files to typescript

SQUASHED MESSAGES:
migrated hooks to typescript

migrated components

migrated apis to ts

migrated views to typescript

migrated views to ts

migrated src files to typescript

cleaning up types

modified commontypes

renamed commonTypes to CommonTypes

### CODE REVIEW

File: src/hooks/useAxiosFunction.ts [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
+}
 
 const useAxiosFunction = () => {
-  const [response, setResponse] = useState([]);
-  const [error, setError] = useState('');
-  const [loading, setLoading] = useState(false);
-  const [controller, setController] = useState();
+  const [response, setResponse] = useState<any[] | any>([]);
+  const [error, setError] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(true);
```
Comment (shashankanandkannan) [inline src/hooks/useAxiosFunction.ts:16]: qq: was switching this from false to true intentional?

**Review by shashankanandkannan — APPROVED** [review]
lgtm

File: src/hooks/useAxiosFunction.ts (hunk 1 above)
Comment (Seanlebon) [reply src/hooks/useAxiosFunction.ts:16]: Probably Not but I think it should work since the useEffect just checks if the value changed

Comment (Seanlebon) [reply src/hooks/useAxiosFunction.ts:16]: def not LOOL

Comment (seanchen0818) [reply src/hooks/useAxiosFunction.ts:16]: Oh wait yeah i think that was, cause we want to start off loading and then set it to not loading when the response is retrieved

**Review by shashankanandkannan — APPROVED** [review]
Approve again 👍

**Pull request merged**